src/main/java/io/logz/jmx2graphite/JavaAgentClient.java

```diff
@@ -0,0 +1,183 @@
+
+    MBeanServer server;
+    ObjectMapper objectMapper;
+
+    public JavaAgentClient() {
+
+        server = ManagementFactory.getPlatformMBeanServer();
+        objectMapper = new ObjectMapper();
+        objectMapper.setVisibility(PropertyAccessor.ALL, JsonAutoDetect.Visibility.NONE);
+        objectMapper.setVisibility(PropertyAccessor.FIELD, JsonAutoDetect.Visibility.ANY);
+    }
+
+    @Override
+    public List<MetricBean> getBeans() throws MBeanClientPollingFailure {
+
+        try {
+            List<MetricBean> metricBeans = Lists.newArrayList();
+            Set<ObjectInstance> instances = server.queryMBeans(null, null);
+
+            for (ObjectInstance instance : instances) {
+                MBeanInfo mBeanInfo = server.getMBeanInfo(instance.getObjectName());
+                List<String> attributes = Lists.newArrayList();
+
+                for (MBeanAttributeInfo attribute : mBeanInfo.getAttributes()) {
+                    attributes.add(attribute.getName());
+                }
+
+                // Dont change to getCanonicalName(), we need it to preserve the order so we can have a valuable metrics tree
+                metricBeans.add(new MetricBean(instance.getObjectName().getDomain() + ":" + instance.getObjectName().getKeyPropertyListString(), attributes));
+            }
+            return metricBeans;
+
+        } catch (IntrospectionException | ReflectionException | InstanceNotFoundException e) {
+            throw new MBeanClientPollingFailure(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public List<MetricValue> getMetrics(List<MetricBean> beans) throws MBeanClientPollingFailure{
+
+        List<MetricValue> metricValues = Lists.newArrayList();
+
+        for (MetricBean metricBean : beans) {
+            try {
+                AttributeList attributeList = server.getAttributes(new ObjectName(metricBean.getName()),
+                                                                   metricBean.getAttributes().toArray(new String[0]));
+
+                Map<String, Object> attrValues = new HashMap<>(attributeList.size());
+                attributeList.asList().forEach((attr) ->
+                        attrValues.put(attr.getName(), attr.getValue()));
+
+                Map<String, Number> metricToValue = flatten(attrValues);
+
+                for (String attrMetricName : metricToValue.keySet()) {
+                    try {
+
+                        long metricTime = LocalDate.now().atStartOfDay(ZoneId.of("UTC")).toEpochSecond();
+
+                        metricValues.add(new MetricValue(
+                                GraphiteClient.sanitizeMetricName(metricBean.getName(), /*keepDot*/ true) + "." + attrMetricName,
+                                metricToValue.get(attrMetricName),
+                                metricTime));
+                    } catch (IllegalArgumentException e) {
+                        logger.info("Can't sent Metric since it's invalid: "+e.getMessage());
```

Comment:
Maybe: Failed converting metric name to Graphite-friendly name: metricsBean.getName  = , attrMetricName =